lib/src/recurrence_rule.dart

```diff
@@ -23,9 +23,9 @@ class RecurrenceRule {
     this.until,
     this.count,
     this.interval,
-    Set<int> bySeconds = const {},
-    Set<int> byMinutes = const {},
-    Set<int> byHours = const {},
+    List<int> bySeconds = const [],
+    List<int> byMinutes = const [],
+    List<int> byHours = const [],
     Set<ByWeekDayEntry> byWeekDays = const {},
```

Comment:
I think that it makes sense to also convert the remaining sets to lists for consistency